packages/preview-middleware-client/src/adp/controllers/AddFragment.controller.ts

```diff
@@ -134,9 +136,9 @@ export default class AddFragment extends BaseDialog {
         const source = event.getSource<Button>();
         source.setEnabled(false);
 
-        const fragmentName = this.model.getProperty('/newFragmentName');
-        const index = parseInt(this.model.getProperty('/selectedIndex'));
-        const targetAggregation = this.model.getProperty('/selectedAggregation/value');
+        const fragmentName = this.model.getProperty('/newFragmentName') as string;
```

Comment:
I am not a big fan of the `as string` everywhere. Could we define it more centrally? Something like https://github.com/SAP/open-ux-tools/compare/feat/1916/reduce-eslint-warnings...feat/1916/reduce-eslint-warnings-proposal?expand=1 ?

Btw, is `selectedIndex` really a string? I think it is a number at least that is what you set in `setProperty`

Reply:
We could also do it purely with typings i.e. we could enforce that the correct type is set by removing the generic `setProperty` function and defining the allowed values per key e.g. 
```typescript
type AddFragmentModel = Omit<JSONModel, 'setProperty'> & {
    getProperty(sPath: '/newFragmentName'): string;
    getProperty(sPath: '/selectedIndex'): string;
    getProperty(sPath: '/selectedAggregation/value'): string;

    setProperty(sPath: '/indexHandlingFlag', value: boolean): void;
};
```